transmogrifier/cli.py

```diff
@@ -42,14 +41,7 @@ def main(source, input_file, output_file, verbose):
     logger.info("Running transform for source %s", source)
 
     transformer = Transformer.load(source, input_file)
-    transformer.write_timdex_records_to_json(output_file)
-    if transformer.processed_record_count == 0:
-        raise ValueError("No records processed from input file, needs investigation")
-    if deleted_records := transformer.deleted_records:
-        deleted_output_file = output_file.replace("index", "delete").replace(
-            "json", "txt"
-        )
-        write_deleted_records_to_file(deleted_records, deleted_output_file)
+    transformer._write_output_files(output_file)
```

Comment:
Some thoughts on `._write_output_files()`.

First of all, I think moving this logic to a method is the right call, so nice work there.

I might also suggest that it should be a **public** method, given it's expected to be called directly from a `Transformer` instance (as opposed to a public method calling private methods not meant for direct usage).

By that same token, I would propose making `write_timdex_records_to_json_file()` and `write_deleted_records_to_txt_file()` **private** methods, as I don't think they should be called directly.

And this brings me to a third point, something I'm pondering in the GeoHarvester as well.  Unceremoniously, but importantly, inside of `write_timdex_records_to_json_file()` there is [this line](https://github.com/MITLibraries/transmogrifier/blob/GDT-83-jsontransformer/transmogrifier/sources/transformer.py#L101) which is actually drivingg the main loop of transformations:
```python
while records:
    ...
```

What I'm not sure is communicated in a method name like `write_output_files()` is that the work of iterating through all input records, transforming them, and _then_ writing is actually getting invoked via this method.

I might suggest that just a small renaming to `transform_and_write_output_files()` would be enough to suggest that not only are files getting written, but this method is kicking off the actual transformation work as well.

Reply:
Agree with all of this, great suggestions!